apps/openmw/mwmechanics/spellcasting.cpp

```diff
@@ -763,6 +763,35 @@ namespace MWMechanics
 
         int school = 0;
 
+        if(mCaster.getClass().isActor()
+                && MWBase::Environment::get().getWorld()->getStore().get<ESM::MagicEffect>()
+                .find(ESM::MagicEffect::WaterWalking))
+        {
+            const MWWorld::CellStore *cell = mCaster.getCell();
```

Comment:
You are checking caster stats, but the caster might not be the recipient of the water walking effect (e.g. with on target spells).